topologytest/topologies.go

```diff
@@ -0,0 +1,314 @@
+// Copyright 2024-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package topologytest
+
+// Topology defines a topology for a set of peers and replications. This can include Couchbase Server, Sync Gateway, and Couchbase Lite peers, with push or pull replications between them.
+type Topology struct {
+	description  string
+	peers        map[string]PeerOptions
+	replications []PeerReplicationDefinition
+}
+
+// Topologies represents user configurations of replications.
+var Topologies = []Topology{
+	{
+		/*
+			+ - - - - - - +
+			' +---------+ '
+			' |  cbs1   | '
+			' +---------+ '
+			' +---------+ '
+			' |   sg1   | '
+			' +---------+ '
+			+ - - - - - - +
+			      ^
+			      |
+			      |
+			      v
+			  +---------+
+			  |   cbl1  |
+			  +---------+
+		*/
+		description: "CBL <-> Sync Gateway <-> CBS",
+		peers: map[string]PeerOptions{
+			"cbs1": {Type: PeerTypeCouchbaseServer, BucketID: PeerBucketID1},
+			"sg1":  {Type: PeerTypeSyncGateway, BucketID: PeerBucketID1},
+			"cbl1": {Type: PeerTypeCouchbaseLiteMock, BucketID: PeerBucketID1},
```

Comment:
I expected this to be PeerTypeCouchbaseLite (and CouchbaseLiteMock would be one possible implementation of that peer).